src/python/TaskWorker/Actions/PreJob.py

```diff
@@ -476,6 +476,26 @@ def touch_logs(self, crab_retry):
             pass
 
 
+    def needsDefer(self):
+        """ Check if the Prejob needs to be deferred, feature useful for some use cases that requires
+            slow release of jobs in a task.
+            The function return True if CRAB_JobReleaseTimeout is defined and not 0, and if the submit
+            time of the task plus the defer time is greater than the current time.
+        """
+        deferTime = int(self.task_ad.get("CRAB_JobReleaseTimeout", 0))
+        if deferTime:
+            self.logger.info('Release timeout specified in extr1aJDL:')
+            totalDefer = deferTime * int(self.job_id)
+            submitTime = int(self.task_ad.get("CRAB_TaskSubmitTime"))
+            currentTime = time.time()
+            if currentTime < (submitTime + totalDefer):
+                self.logger.info('  Release timeout specified in extr1aJDL: deferring %s seconds' % totalDefer)
+                return True
+            else:
+                self.logger.info('  Continuing normally since current time is greater than foreseen starttime of the job')
```

Comment:
misleading message, better: "current time is greater than requested starttime of the job"